plexapi/livetv.py

```diff
@@ -0,0 +1,232 @@
+        self.key = data.attrib.get('key', '')
+        self.uuid = data.attrib.get('uuid')
+        self.uri = data.attrib.get('uri')
+        self.protocol = data.attrib.get('protocol')
+        self.status = data.attrib.get('status')
+        self.state = data.attrib.get('state')
+        self.lastSeenAt = utils.toDatetime(data.attrib.get('lastSeenAt'))
+        self.make = data.attrib.get('make')
+        self.model = data.attrib.get('model')
+        self.modelNumber = data.attrib.get('modelNumber')
+        self.source = data.attrib.get('source')
+        self.sources = data.attrib.get('sources')
+        self.thumb = data.attrib.get('thumb')
+        self.tuners = utils.cast(int, data.attrib.get('tuners'))
+        self.items = self.findItems(data)
+
+
+@utils.registerPlexObject
+class DVR(DVRDevice):
+    """ Represents a single DVR."""
+
+    TAG = 'Dvr'
+
+    def _loadData(self, data):
+        self._data = data
+        self.key = utils.cast(int, data.attrib.get('key'))
+        self.uuid = data.attrib.get('uuid')
+        self.language = data.attrib.get('language')
+        self.lineupURL = data.attrib.get('lineup')
+        self.title = data.attrib.get('lineupTitle')
+        self.country = data.attrib.get('country')
+        self.refreshTime = utils.toDatetime(data.attrib.get('refreshedAt'))
+        self.epgIdentifier = data.attrib.get('epgIdentifier')
+        self.items = self.findItems(data)
+
+
+class LiveTV(PlexObject):
+    def __init__(self, server, data, session=None, token=None):
+        self._token = token
+        self._session = session or requests.Session()
+        self._server = server
+        self.dvrs = []  # cached DVR objects
+        super().__init__(server, data)
+
+    def _loadData(self, data):
+        """ Load attribute values from Plex XML response. """
+        self._data = data
+        self.cloud_key = data.attrib.get('machineIdentifier')
+
+    def _get_cloud_key(self):
+        url = self._server.url(key='/tv.plex.providers.epg.cloud', includeToken=True)
+        data = requests.get(url=url).json()
+        if data:
+            self.cloud_key = data.get('MediaContainer').get('Directory')[1].get('title')
+            return self.cloud_key
+        return None
+
+    def dvrs(self):
+        """ Returns a list of :class:`~plexapi.livetv.DVR` objects available to your server.
+        """
+        if not self.dvrs:
+            self.dvrs = self.fetchItems('/livetv/dvrs')
+        return self.dvrs
+
+    def sessions(self):
+        """ Returns a list of all active live tv session (currently playing) media objects.
+        """
+        return self.fetchItems('/livetv/sessions')
+
+    def directories(self):
+        """ Returns a list of all :class:`~plexapi.livetv.Directory` objects available to your server.
+        """
+        return self._server.fetchItems(self.cloud_key + '/hubs/discover')
+
+    def _guide_items(self, grid_type: int, beginsAt: int = None, endsAt: int = None):
```

Comment:
Imo it would be much better of beginsAt and endsAt was datetime and not it, we can convert it later, but it makes it more user friendly imo

Reply:
Agreed. I'm pretty sure there's already a method in here somewhere to convert datetime and int